components/video-page/short-form-panel.tsx

```
@@ -0,0 +1,681 @@
+    const payload = await response.json().catch(() => null);
+    if (!response.ok)
+      throw new Error(readClientApiError(payload, 'Failed to load short-form batches'));
+    const next = (payload.data?.batches ?? []) as Batch[];
+    setAvailable(true);
+    setCanEdit(Boolean(payload.data?.canEdit));
+    setBatches(next);
+    if (next[0]) await loadBatch(next[0].id);
+  }, [loadBatch, roughCutId]);
+
+  useEffect(() => {
+    void loadBatches().catch((cause) =>
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/components/video-page/short-form-panel.tsx:103:10
  101 |
  102 |   useEffect(() => {
> 103 |     void loadBatches().catch((cause) =>
      |          ^^^^^^^^^^^ Avoid calling setState() directly within an effect
  104 |       setError(cause instanceof Error ? cause.message : String(cause))
  105 |     );
  106 |   }, [loadBatches]);

(react-hooks/set-state-in-effect)